custom-rst-roles/__init__.py

```diff
@@ -10,7 +10,28 @@ def spellexception_role(
     return [node], []
 
 
+def literalref_role(
+    name, rawtext, text, lineno, inliner, options=None, content=None
+):
+
+    if "<" in text and ">" in text:
+        linktext = text.split("<")[0].strip()
+        linkurl = text.split("<")[1].split(">")[0].strip()
+    else:
+        linktext = text
+        linkurl = text
+
+    if not linkurl.startswith("http"):
```

Comment:
Changed it to just check for `://`, so that links to for example `ftp://something` will still work.
But we need to add http by default - otherwise, you cannot just write ``:literalref:`ubuntu.com` ``, but would need to do ``:literalref:`ubuntu.com <https://ubuntu.com>` `` just for a simple URL.